### DISCUSSION TITLE
Suggestion to rename mingw's i386 target to i686 for Xmake 3

### DISCUSSION THREAD

**Discussion by ifarbod** [xmake-io/xmake#5904] [Ideas]
Xmake 3 introduces breaking changes, right?

So why not rename the MinGW target to **i686** as it's exclusively known as that? And maybe provide a backup option for people who still use `i386`.

![VSCode Xmake screenshot](https://github.com/user-attachments/assets/5a23e202-317d-4467-ab0b-d40d30aaba45)

Some references:

https://github.com/msys2/MSYS2-packages/blob/ecb03e70dc7ac85e2d2fffa015e4fbfd3308d378/fish/msystem.fish#L19

https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/MinGW.cpp#L648

https://github.com/JuliaLang/julia/blob/master/test/binaryplatforms.jl#L171

**Comment by waruqi** [comment]
You can mention it at https://github.com/xmake-io/xmake/issues/5527

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake/discussions/5906

**Comment by jingkaimori** [comment]
usage of i386 for mingw plat:

- https://github.com/ssbssa/gdb/blob/22b96f176da2e42a73bb665c5578be3d98c9de43/gas/NEWS#L350

**Comment by TurnOffNOD** [comment]
i686 means that compiled binary should run at least 686 architecture, and means it drops out suport of architecture older than 686.

i386 means that it still supports architecture older than 686.

**Comment by waruqi** [comment]
I added a poll option, x86_64, i386, i686

https://github.com/xmake-io/xmake/discussions/5906
